src/planner/rules/schema.rs

```diff
@@ -0,0 +1,104 @@
+//! Analyze schema and replace all column references with physical indices.
+//!
+//! This is the final step before executing.
+
+use super::*;
+use crate::types::ColumnIndex;
+
+/// Replaces all column references (`ColumnRefId`) with
+/// physical indices ([`ColumnIndex`]) to the given schema.
+///
+/// # Example
+/// - given schema:           `sum(v1), v2`
+/// - the expressions:        `v2 + 1, sum(v1) + v2`
+/// - should be rewritten to: `#1 + 1, #0 + #1`
+pub fn resolve_column_index(expr: RecExpr, schema: &[RecExpr]) -> RecExpr {
```

Comment:
This is exactly what I want to do for a long time. We should not have column index until the very last stage of planning.

Reply:
++++1